cmd/cli/tanzu/main.go

```diff
@@ -13,7 +14,14 @@ import (
 
 func main() {
 	if err := core.Execute(); err != nil {
-		fmt.Println("")
-		log.Fatal(err)
+		if errStr, ok := err.(*exec.ExitError); ok {
+			// If a plugin exited with an error, we don't want to print its
```

Comment:
It's not just the plugin but applies to all commands. Plugin is also another cobra command which we have some mechanism written to invoke as a separate binary.
So, Basically Tanzu CLI is a combination of its own commands like `tanzu plugin` and some plugins like `tanzu cluster`. We might want to use command as a generic word to include all core commands as well as plugin invocation. This also applies to other comments as well as PR title and description.

```suggestion
			// If a command exited with an error, we don't want to print its
```

Reply:
Thanks for the clarification @anujc25!

In this particular case, IIUC, it is just plugins that will exit with an error of type `exec.ExitError` and provide an exit code because the plugin command calls the plugin as an external process.
For native cobra commands of the tanzu cli (like `tanzu plugin delete`) an error will be returned but of a different type and fall into the `else` case.

I tried to test both code paths by using `tz cluster kubeconfig get` to generate an error from a plugin and `tz plugin delete` to generate an error from a command.

Does that make sense, or did I mis-understand?

Reply:
Ahh. This makes sense. Thanks for the explanation.